roles/installer/tasks/main.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: Check that necessary variables are defined
+  assert:
+    that:
+      - installer_scenario is defined
+
+- name: Check if initial install has happened
+  stat:
+    path: /etc/foreman-installer/scenarios.d/.installed
+  register: installed
+
+- name: 'Check if changes are needed by running installer in noop mode'
+  command: >
+    {{ installer_command }}
+    --scenario {{ installer_scenario }}
+    --noop
+  register: installer_noop
+  changed_when: false
+  failed_when: installer_noop.rc not in [0, 2]
+  when: installed.stat.exists
+
+- name: 'Run installer' # noqa 301
```

Comment:
What's 301?

Reply:
Good reminder, that was due to pre-idempotence, now that I have that I can remove this.